### ISSUE TITLE
Request: Forward stderr to terminal when capture:true is used

### ISSUE THREAD

**Issue by mid-kid** [mesonbuild/meson#13332]
When using `capture: true` on `custom_target` or `generator`, [both the `stdout` and `stderr` are captured](https://github.com/mesonbuild/meson/blob/e991c4d45460c09e5eaee6cb1041edb95d1e990e/mesonbuild/scripts/meson_exe.py#L58) and `stdout` is redirected to a file.

However, I'm running a program that outputs useful information, like warnings, to `stderr`, even if the program exits successfully.
I think that it'd be a good idea to always forward the stderr to the terminal, as this is usually expected behavior of any program that only consumes the stdout (ideally `capture: true` would just be the equivalent of `./prog > file` in a makefile)

**Comment by bonzini** [comment]
I think it would make sense at least to print it if the program fails and `strict` is true

**Comment by JamBalaya56562** [comment]
Thank you for the request — this has been implemented. With `capture: true`, Meson now forwards a wrapped program's stderr to the terminal even when the program exits successfully: stdout is still redirected to the capture file, while stderr (warnings and other diagnostics) reaches the console.

In `mesonbuild/scripts/meson_exe.py`, `run_exe()` now does, after a successful run:

```python
elif stderr:  # Allow a wrapped subprocess to gracefully communicate warnings
    encoding = locale.getpreferredencoding()
    print(stderr.decode(encoding=encoding, errors='replace'))
```

This matches the requested `./prog > file` semantics, where stderr stays on the terminal.

**Verification:** introduced in commit [`c5618e3f`](https://github.com/mesonbuild/meson/commit/c5618e3f5001dcf41e5c3d4c83a00e8c454f0800) ("meson_exe: Always emit stderr when it has content"); current code in `mesonbuild/scripts/meson_exe.py`. (The implementation is actually a bit broader than the "fail + strict only" suggestion in the thread — stderr with content is now always surfaced.)

This looks safe to close. Thanks for the request.

**Comment by mid-kid** [comment]
The AI sludge was unnecessary, a simple link would've sufficed: https://github.com/mesonbuild/meson/pull/15683

Thanks for letting me know, though.

**Comment by bonzini** [comment]
Yeah @JamBalaya56562 is evidently using AI to triage all the existing Meson issues... It is a bit wordy but the report does help a bit in double checking for mistakes of the AI (not many so far but >0). I will take whatever help I get, considering it does save time in the exciting task of triaging of ten years of issues (that is I am not closing them mindlessly).
